ner_v2/detectors/numeral/number/en/number_detection.py

```diff
@@ -1,262 +1,50 @@
-            corresponding substrings in the original message.
-
-            For example:
-                (['3'], ['for 3 people'])
-
-        """
-        number_list, original_list = self.task_dict.get(self.entity_name, self.task_dict['Default'])()
-        self._update_processed_text(original_list)
-        return number_list, original_list
-
-    def _detect_number_format(self):
-        """Detects any numbers from text
-        This is a default function which will be called when we want to detect the number from the text
-
-        Returns:
-            A tuple of two lists with first list containing the detected numbers and second list containing their
-            corresponding substrings in the original message.
-
-            For example:
-                input: "I want to purchase 30 units of mobile and 40 units of Television"
-                output: (['30', '40'], ['30','40'])
-
-        Note: By default we detect numbers from 1 digit to 3 digit
-        """
-        number_list = []
-        original_list = []
-        patterns = re.findall(r'\s([0-9]{{{0},{1}}})[\s|,]'.format(self.min_digit, self.max_digit),
-                              self.processed_text.lower())
-        for pattern in patterns:
-            number_list.append(pattern)
-            original_list.append(pattern)
-        word2number_number_list, word2number_original_list = self._detect_numerals()
-        number_list = number_list + word2number_number_list
-        original_list = original_list + word2number_original_list
-        return number_list, original_list
-
-    def _detect_number_of_people_format(self):
+    def _custom_detect_number_of_people_format(self, number_list=None, original_list=None):
         """Detects any numbers from text
         This is a function which will be called when we want to detect the number of persons from the text
 
         Returns:
             A tuple of two lists with first list containing the detected numbers and second list containing their
             corresponding substrings in the original message.
 
-            For example:
+            For example:9
                 input text: "Can you please help me to book tickets for 3 people"
                 output: (['3'], ['for 3 people'])
 
         Note: Currently, we can detect numbers from 1 digit to 3 digit. This function will be enabled only
         if entity_name is set to  'number_of_people'
         """
-        number_list = []
-        original_list = []
-        patterns = re.findall(
-            r'\s((fo?r)*\s*([0-9]{{{0},{1}}})\s*(ppl|people|passengers?|travellers?|persons?|pax|adults?)*)\s'.format(
-                self.min_digit, self.max_digit),
-            self.processed_text.lower()
-        )
+        number_list = number_list or []
+        original_list = original_list or []
+        patterns = re.findall(r'\s((fo?r)*\s*([0-9]+)\s*(ppl|people|passengers?|travellers?|persons?|pax|adults?))\s',
```

Comment:
Make for and whitespace optional instead of `*`
`\s((fo?r\s+)?([0-9]+)\s*(ppl|people|passengers?|travellers?|persons?|pax|adults?))\s`

Reply:
Made changes